streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImpl.java

```diff
@@ -119,15 +121,43 @@ public final void removeAllTaskLevelSensors(final String taskName) {
         }
     }
 
+    public Sensor nodeLevelSensor(final String taskName, final String processorNodeName, final String sensorName, final Sensor.RecordingLevel recordingLevel, final Sensor... parents) {
+        final String key = threadName + "." + taskName + ".node." + processorNodeName;
+        synchronized (nodeLevelSensors) {
+            if (!nodeLevelSensors.containsKey(key)) {
+                nodeLevelSensors.put(key, new LinkedList<>());
+            }
+
+            final String fullSensorName = key + "." + sensorName;
+
+            final Sensor sensor = metrics.sensor(fullSensorName, recordingLevel, parents);
+
+            nodeLevelSensors.get(key).push(fullSensorName);
+
+            return sensor;
+        }
+    }
+
+    public final void removeAllNodeLevelSensors(final String taskName, final String processorNodeName) {
+        final String key = threadName + "." + taskName + ".node." + processorNodeName;
```

Comment:
this is the same as above: we should extract to a method to avoid inconsistencies if one might get updated but the other one slips...